src/components/database/DatabaseOverview.vue

```diff
@@ -0,0 +1,250 @@
+        return {
+            name: name,
+            type: showFilterType(filter.filter),
+            taxaCount: taxonCounts.value.get(name) || 'computing...',
+            proteinCount: proteinCounts.value.get(name) || 'computing...',
+        };
+    });
+});
+
+const showFilterType = (filterType: FilterType) => {
+    switch (filterType) {
+        case FilterType.UniProtKB:
+            return 'N/A';
+        case FilterType.Taxon:
+            return 'Taxa';
+        case FilterType.Protein:
+            return 'Proteins';
+        case FilterType.Proteome:
+        default:
+            return 'Reference Proteomes';
+    }
+};
+
+const computeTaxonCount = async (filter: Filter) => {
+    switch (filter.filter) {
+        case FilterType.UniProtKB:
+            const total = await taxonomyCommunicator.getResponses([ 1 ]);
+            return total.reduce((acc, taxonomy) => acc + taxonomy.descendants.length, 0);
+        case FilterType.Taxon:
+            const taxonomies = await taxonomyCommunicator.getResponses(filter.data);
+            return taxonomies.reduce((acc, taxonomy) => acc + taxonomy.descendants.length, 0);
+        case FilterType.Protein:
+            await updateProteinOntology(filter.data);
+            return new Set<number>(filter.data?.map(p => proteinOntology.get(p)?.taxonId)).size
+        case FilterType.Proteome:
+        default:
+            await updateProteomeOntology(filter.data);
+            return new Set<number>(filter.data?.map(p => proteomeOntology.get(p)?.taxonId)).size
+    }
+};
+
+const computeProteinCount = async (filter: Filter) => {
+    switch (filter.filter) {
+        case FilterType.UniProtKB:
+            return await proteinCommunicator.getProteinCount();
+        case FilterType.Taxon:
+            return await UniprotCommunicator.getRecordCount(filter.data);
+        case FilterType.Protein:
+            return filter.data.length;
+        case FilterType.Proteome:
+        default:
+            await updateProteomeOntology(filter.data);
+            return filter.data.reduce((acc, proteome) => acc + proteomeOntology.get(proteome)?.proteinCount || 0, 0);
+    }
+};
+
+const tableHeaders = [
+    { title: 'Name', key: 'name' },
+    { title: 'Type', key: 'type' },
+    { title: '# Taxa', key: 'taxaCount' },
+    { title: '# Proteins', key: 'proteinCount' },
+    { title: 'Actions', key: 'actions', sortable: false },
+];
+
+const editDatabase = (name: string) => {
+    databaseToManipulate.value = name;
+    editDatabaseDialogOpen.value = true;
+};
+
+const confirmEditDatabase = async (name: string, filter: Filter) => {
+    emits('database:update', databaseToManipulate.value, name, filter);
+
+    const taxonCount = await computeTaxonCount(filter);
+    const proteinCount = await computeProteinCount(filter);
+    taxonCounts.value.set(name, `~ ${formatNumber(taxonCount)}`);
+    proteinCounts.value.set(name, `~ ${formatNumber(proteinCount)}`);
```

Comment:
Since we're really specific about the amount of taxa and proteins that are part of a database, we could remove the `~`.

Reply:
Won't reference proteomes still have an overlap for the amount of proteins?

Reply:
Ah yes, indeed. You can ignore my comment then :P